lib/events/filelog.go

```diff
@@ -222,6 +222,84 @@ func (l *FileLog) SearchUnstructuredEvents(ctx context.Context, req SearchEvents
 	return evts, next, nil
 }
 
+// checkpointKey is a pointer to an event in the log. This is used to keep track
+// of the last processed event so the caller can resume querying from this event.
+type checkpointKey struct {
+	id        string
+	timestamp time.Time
+}
+
+// MarshalText marshals a checkpointKey into plain text.
+func (c checkpointKey) MarshalText() ([]byte, error) {
+	if c.id == "" {
+		return []byte{}, nil
+	}
+	if c.timestamp.IsZero() {
+		return []byte(c.id), nil
+	}
+	return []byte(c.id + "/" + c.timestamp.Format(time.RFC3339)), nil
```

Comment:
This is some timestamp that's marshaled and unmarshaled by computers, can we make it the UnixNano number in decimal or hex instead? I'm not sure that the truncation (which we could get rid of by using `time.RFC3339Nano`, to be fair) or the tz offset will cause problems.